neuraxle/metaopt/auto_ml.py

```diff
@@ -0,0 +1,358 @@
+        :class:`HyperparameterSamples`,
+        :class:`HyperparameterSpace`
+    """
+
+    def __init__(self, validation_technique=None):
+        AutoMLStrategyMixin.__init__(self, validation_technique)
+        MetaStepMixin.__init__(self, None)
+        BaseStep.__init__(self)
+
+    def find_next_best_hyperparams(self, trials_data_container: 'AutoMLDataContainer') -> HyperparameterSamples:
+        return trials_data_container.hyperparameter_space.rvs()
+
+
+class AutoMLDataContainer:
+    """
+    Simple data container used by :class:`AutoMLStrategyMixin`, and :class:`AutoMLSequentialWrapper`.
+    """
+
+    def __init__(
+            self,
+            trial_number: int,
+            scores: List[float],
+            hyperparams: List[HyperparameterSamples],
+            hyperparameter_space: HyperparameterSpace,
+            n_iters: int
+    ):
+        self.hyperparams = hyperparams
+        self.scores = scores
+        self.trial_number = trial_number
+        self.hyperparameter_space = hyperparameter_space
+        self.n_iters = n_iters
+
+
+class AutoMLSequentialWrapper(NonTransformableMixin, MetaStepMixin, BaseStep):
+    """
+    A step to execute any Automatic Machine Learning Algorithms.
+
+    Example usage :
+
+    .. code-block:: python
+
+        auto_ml = AutoMLSequentialWrapper(
+            auto_ml_strategy=RandomSearchAutoMLStrategy(),
+            step=ForecastingPipeline(),
+            hyperparams_repository=HyperparamsJSONRepository(),
+            n_iters=100
+        )
+
+        auto_ml: AutoMLSequentialWrapper = auto_ml.fit(data_inputs, expected_outputs)
+
+        best_model: ForecastingPipeline = auto_ml.get_best_model()
+
+
+    .. seealso::
+        :class:`AutoMLStrategyMixin`,
+        :class:`RandomSearchAutoMLStrategy`,
+        :class:`HyperparamsRepository`,
+        :class:`MetaStepMixin`,
+        :class:`BaseStep`
+    """
+
+    def __init__(
+            self,
+            auto_ml_strategy: AutoMLStrategyMixin,
+            step: BaseStep,
+            hyperparams_repository: HyperparamsRepository,
+            n_iters: int
+    ):
+        NonTransformableMixin.__init__(self)
+
+        auto_ml_strategy = auto_ml_strategy.set_step(step)
+        MetaStepMixin.__init__(self, auto_ml_strategy)
+
+        self.hyperparams_repository = hyperparams_repository
+        self.n_iters = n_iters
+
+    def fit(self, data_inputs, expected_outputs=None) -> BaseStep:
+        """
+        Find the best hyperparams using the wrapped AutoML strategy.
+
+        :param data_inputs: data inputs
+        :param expected_outputs: expected ouptuts to fit on
+        :return: fitted self
+        :rtype: BaseStep
+        """
+        for i in range(self.n_iters):
+            auto_ml_trial_data_container = self._load_auto_ml_data(i)
+
+            hyperparams = self.wrapped.find_next_best_hyperparams(auto_ml_trial_data_container)
+            self.wrapped = self.wrapped.set_hyperparams(hyperparams)
```

Comment:
Here we need something similar to the RandomSearch's code : 
1. copy a self.virgin_wrapped
2. set params on the copy
3. train the copy and then discard it. 

Note: this makes me think that we might also like an option to save each models' full dump somewhere, not only their hyperparameters. 

Other important comment: 
Replace `self.wrapped = self.wrapped.set_hyperparams(hyperparams)` by `self.wrapped = self.wrapped_virgin.update_hyperparams(hyperparams)` because we'd like to allow the user to omit specifying all the hyperparameter distributions. E.g.: have 10 hyperparams by default, and only 1 distribution defined in the space.

Reply:
already done in AutoMLAlgorithm class : 

1. copy a self.virgin_wrapped
2. set params on the copy

Reply:
@guillaume-chevalier that made me think... why are we not using a copy.deepcopy in random search ? that could cause problems... any changes in previous trials should not reflect on the current trial

Reply:
added refit, and fixed deep copy issues